geowebcache/s3storage/src/main/java/org/geowebcache/s3/S3BlobStoreInfo.java

```diff
@@ -334,7 +337,38 @@ public BlobStore createInstance(TileLayerDispatcher layers, LockProvider lockPro
         checkState(
                 isEnabled(),
                 "Can't call S3BlobStoreConfig.createInstance() is blob store is not enabled");
-        return new S3BlobStore(this, layers, lockProvider);
+        final GeoWebCacheEnvironment gwcEnvironment =
+                GeoWebCacheExtensions.bean(GeoWebCacheEnvironment.class);
+        return new S3BlobStore(this.clone(gwcEnvironment, true), layers, lockProvider);
+    }
+
+    public S3BlobStoreInfo clone(
+            GeoWebCacheEnvironment gwcEnvironment, Boolean allowEnvParametrization) {
+        S3BlobStoreInfo blobStore = SerializationUtils.clone(this);
+
+        if (allowEnvParametrization && gwcEnvironment != null) {
+            blobStore.setName(getName());
+            blobStore.setEnabled(isEnabled());
+            blobStore.setDefault(isDefault());
+            blobStore.setAccess(getAccess());
+            blobStore.setPrefix(getPrefix());
+            blobStore.setUseHTTPS(isUseHTTPS());
+            blobStore.setUseGzip(isUseGzip());
```

Comment:
I see these cannot be parameterized... on one side, they seem less likely to be parameterized, on the other end, while you're here, why not do them all?